Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp

```diff
@@ -6243,20 +6243,28 @@ bool CStaticFunctionDefinitions::SetTime(unsigned char ucHour, unsigned char ucM
 }
 
 bool CStaticFunctionDefinitions::ProcessLineOfSight(const CVector& vecStart, const CVector& vecEnd, bool& bCollision, CColPoint** pColPoint,
-                                                    CClientEntity** pColEntity, const SLineOfSightFlags& flags, CEntity* pIgnoredEntity,
+                                                    CClientEntity** pColEntity, const SLineOfSightFlags& flags, std::vector<CClientEntity*> vecIgnoredElements,
                                                     SLineOfSightBuildingResult* pBuildingResult)
 {
     assert(pColPoint);
     assert(pColEntity);
 
-    if (pIgnoredEntity)
-        g_pGame->GetWorld()->IgnoreEntity(pIgnoredEntity);
+    std::vector<CClientEntity*> vecDidIgnoreElements;
+
+    for (CClientEntity* pIgnoredElement : vecIgnoredElements)
+    {
+        if (CStaticFunctionDefinitions::GetElementCollisionsEnabled(*pIgnoredElement))
+        {
+            CStaticFunctionDefinitions::SetElementCollisionsEnabled(*pIgnoredElement, false);
+            vecDidIgnoreElements.push_back(pIgnoredElement);
+        }
+    }
```

Comment:
You shouldn't be creating a new vector here, it's slow.
I'd suggest you to do this instead:
```cpp
vecIgnoredElements.erase(std::remove_if(vecIgnoredElements.begin(), vecIgnoredElements.end(), [](const CClientEntity* pEntity) {
    // Remove entities that already have their colision disabled. 
    // This prevents us from reenabling them.
    if (CStaticFunctionDefinitions::GetElementCollisionsEnabled(*pIgnoredElement))
        return true;
    
    // Otherwise disable collision and keep it in the array
    CStaticFunctionDefinitions::SetElementCollisionsEnabled(*pIgnoredElement, false);
    
    return false;       
), vecIgnoredElements.end();
```